update_sample_status/main.py

```diff
@@ -1,63 +1,120 @@
+        filename = external_id + file_extension
+        md5_blob = bucket.get_blob(filename + '.md5')
+        primary_blob = bucket.get_blob(filename)
+        secondary_files = [external_id + f for f in output_type['secondary']]
+
+        # Create the metadata object for the gvcf and for the cram.
+        file_object = {
+            'location': join('gs://', upload_bucket, filename),
+            'basename': filename,
+            'class': 'File',
+            'checksum': md5_blob.download_as_string().decode(),
+            'size': primary_blob.size,
+            'secondaryFiles': secondary_files,
+            'batch': batch,  # TODO: Check if this is ok here, or if it should be stored elsewhere.
+        }
+
+        file_objects.append(file_object)
+
+    return file_objects
+
+
+@routes.put('/upload_sample')
+async def upload_sample(request):
+    """ Registers file upload in Sample Metadata DB """
+
+    sapi = SampleApi()
+
+    request_json = await request.json()
+    qc_metadata = request_json.get('metadata')
+    external_id = request_json.get('sample')
     project = request_json.get('project')
-    sample = request_json.get('sample')
+    batch = str(request_json.get('batch'))
     status = request_json.get('status')
-    if not project or not sample or not status:
-        return abort(400)
-
-    logging.info(f'Processing request: {request_json}')
-
-    # Fetch the per-project configuration from the Secret Manager.
-    gcp_project = os.getenv('GCP_PROJECT')
-    secret_name = (
-        f'projects/{gcp_project}/secrets' '/update-sample-status-config/versions/latest'
-    )
-    config_str = secret_manager.access_secret_version(
-        request={'name': secret_name}
-    ).payload.data.decode('UTF-8')
-    config = json.loads(config_str)
-
-    project_config = config.get(project)
-    if not project_config:
-        return abort(404)
-
-    # Get the Airtable credentials.
-    base_key = project_config.get('baseKey')
-    table_name = project_config.get('tableName')
-    api_key = project_config.get('apiKey')
-    if not base_key or not table_name or not api_key:
-        return abort(500)
-
-    # Update the entry.
-    airtable = Airtable(base_key, table_name, api_key)
-    try:
-        response = airtable.update_by_field('Sample ID', sample, {'Status': status})
-    except HTTPError as err:  # Invalid status enum.
-        logging.error(err)
-        return abort(400)
-
-    if not response:  # Sample not found.
-        return abort(404)
-
-    return ('', 204)
+
+    if not project or not external_id or not status or not batch or not qc_metadata:
+        raise web.HTTPBadRequest(
+            reason=f'Invalid request. Ensure that the project, sample, status, batch and metadata are provided.'
+        )
+
+    # Determine the internal ID
+    external_id_dict = {'external_ids': [external_id]}
+    internal_id_map = sapi.get_sample_id_map_by_external(project, external_id_dict)
```

Comment:
This recently changed, so you don't need to wrap it in a dictionary anymore.

```suggestion
    internal_id_map = sapi.get_sample_id_map_by_external(project, [external_id])
```